### PR TITLE
Implementation of OCB mode

### LINKED ISSUE #788: Support for OCB cipher mode of operation
OCB is an AEAD scheme that is specified in [RFC7253](https://tools.ietf.org/html/rfc7253).

The author made a mode friendly specification in https://web.cs.ucdavis.edu/~rogaway/ocb/ocb-faq.htm

### PR DESCRIPTION
**Pull request by aberenguel**
Fixes #788

Checklist:
- [x] Correct file headers (see CONTRIBUTING.md).
- [x] Formatted with [SwiftFormat](https://github.com/nicklockwood/SwiftFormat).
- [x] Tests added.

Changes proposed in this pull request:
- Implementation of OCB mode (RFC7253)

### CODE REVIEW

**Comment by aberenguel** [comment]
I had to change the BlockMode protocol because the OCB needs both cipherOperations in decryption: encrypt and decrypt.

File: Tests/CryptoSwiftTests/AESOCBTests.swift [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    let P: Array<UInt8>
+    let C: Array<UInt8>
+  }
+
+  func testAESOCBWithRFC7253Tests() {
+
+    var fixtures = [
+      TestFixture(K: Array<UInt8>(hex: "000102030405060708090A0B0C0D0E0F"),
+                  N: Array<UInt8>(hex: "BBAA99887766554433221100"),
+                  P: Array<UInt8>(hex: ""),
```
Comment (aberenguel) [inline Tests/CryptoSwiftTests/AESOCBTests.swift]: This fixture revealed a bug: if the ciphertext is empty (there is only a tag), the tag is not verified

Comment (aberenguel) [reply Tests/CryptoSwiftTests/AESOCBTests.swift]: Fixed!

**Comment by krzyzanowskim** [comment]
thank you very much. I'll review it more thoroughly as soon as possible. I wonder if we can do this without adding parameter to `BlockMode` function. The `cipherOperation` was intended to be either encryption or decryption

**Comment by krzyzanowskim** [comment]
@aberenguel I'd like to get this one in for the next release. Can you please resolve the merge conflict?

**Pull request merged**